### ISSUE TITLE
workload/{bank,insights}: tests non-deterministically fail to split ranges when pointed to a secondary tenant

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#109458]
**Describe the problem**

after #108762, the tests in pkg/workload/bank and /insights fail non-terministically (e.g. under stress) when pointed to a secondary tenant:

```
=== RUN   TestInsightsWorkload/rows=10/ranges=9
    insights_test.go:74: got 1 ranges expected 9
=== RUN   TestInsightsWorkload/rows=10/ranges=10
    insights_test.go:74: got 1 ranges expected 10
=== RUN   TestInsightsWorkload/rows=10/ranges=100
    insights_test.go:74: got 1 ranges expected 10
=== CONT  TestInsightsWorkload
    insights_test.go:78: -- test log scope end --
test logs left over in: /tmp/_tmp/eba056330b249b393168d116b1921cec/logTestInsightsWorkload3333307126
--- FAIL: TestInsightsWorkload (2.85s)
    --- PASS: TestInsightsWorkload/rows=10/ranges=0 (0.07s)
    --- PASS: TestInsightsWorkload/rows=10/ranges=1 (0.20s)
    --- FAIL: TestInsightsWorkload/rows=10/ranges=9 (0.19s)
    --- FAIL: TestInsightsWorkload/rows=10/ranges=10 (0.19s)
    --- FAIL: TestInsightsWorkload/rows=10/ranges=100 (0.19s)
```

Same error for:
- `TestBankWorkload` in workload/bank
- `TestSplits` in workloadsql

**Expected behavior**

The test should deterministically succeed.

Epic: CRDB-26687

Jira issue: CRDB-30935

**Comment by knz** [comment]
@stevendanna suggests extracing the core logic of this test into a separate test that will make the investigation easier. The simpler test would do:

1. create a table (somewhat like the bank workload)
2. split it (using logic similar to workload.Split)
3. use SHOW RANGES to ascertain the range count

Then stress that.
Upon a failure, we may want to look at the logs.

Three areas to look at:
- is the Split() call actually doing a split? (check in range logs after the call?)
- is SHOW RANGES actually reporting current state? (check output of show ranges vs range descriptors)
- consider removing the special case error handling in workload.Split

**Comment by yuzefovich** [comment]
I think that this might have been fixed by #109449 - from one the failures Raphael observed https://teamcity.cockroachdb.com/viewLog.html?buildId=11470530&buildTypeId=Cockroach_BazelEssentialCi
```
I230824 10:33:28.339062 4533 5@util/log/event_log.go:32  [T10,Vtenant-10,nsql1,client=127.0.0.1:53576,hostssl,user=root] 369 ={"Timestamp":1692873208330733899,"EventType":"create_table","Statement":"CREATE TABLE test.public.insights_workload_table_0 (id INT8 PRIMARY KEY, balance DECIMAL NOT NULL, payload STRING NOT NULL, shared_key INT8 NOT NULL)","Tag":"CREATE TABLE","User":"root","DescriptorID":108,"TableName":"test.public.insights_workload_table_0"}
I230824 10:33:28.360993 4986 workload/workloadsql/workloadsql.go:126  [-] 370  skipping workload splits; can't scatter on tenants'
```

**Comment by yuzefovich** [comment]
Indeed, confirmed that it's no longer reproducible on master but quickly fails when #109449 is reverted, will send unskips.
